python/src/aiconfig/editor/server/server.py

```diff
@@ -226,3 +226,18 @@ def delete_prompt() -> FlaskResponse:
 
     operation = make_op_run_method(method_name)
     return run_aiconfig_operation_with_request_json(aiconfig, request_json, f"method_{method_name}", operation, signature)
+
+
+@app.route("/api/update_model", methods=["POST"])
+def update_model() -> FlaskResponse:
+    state = get_server_state(app)
+    aiconfig = state.aiconfig
+    request_json = request.get_json()
+
+    model_name: str | None = request_json.get("model_name")
+    settings: Dict[str, Any] | None = request_json.get("settings")
+    prompt_name: str | None = request_json.get("prompt_name")
+
+    operation = make_op_run_method(MethodName("update_model"))
+    operation_args: Result[OpArgs, str] = result.Ok(OpArgs({"model_name": model_name, "settings": settings, "prompt_name": prompt_name}))
+    return run_aiconfig_operation_with_op_args(aiconfig, "update_model", operation, operation_args)
```

Comment:
Looks good, please just make sure to address any pyright errors. It actually catches a lot of problems like if you accidentally treated a Result[T] like a T. 

BTW, for miscellaneous reasons I have been suppressing pyright errors judiciously and making sure pyright reports exactly 0.

Example of a fixme (this like a P1 todo): https://github.com/lastmile-ai/aiconfig/blob/526d45e773577eb27e0abc33b87fb5df144f246c/python/src/aiconfig/eval/common.py#L133

In this case my data model is actually broken so I called it a fixme.

Another example: https://github.com/lastmile-ai/aiconfig/blob/526d45e773577eb27e0abc33b87fb5df144f246c/python/src/aiconfig/eval/metrics.py#L88

In this case it's just "ignore" because the underlying library is untyped, nothing can be done here.